iothub/device/src/AdditionalClientInformation.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    // Holder for additional client properties such as product information, PnP model Id.
+    // This class does not hold any client credential information.
+    internal class AdditionalClientInformation
```

Comment:
This is a holder for additional client information that is passed on to the transport layer. Previously we were passing in the complete client options but this now exposes only the specific required information.